### DISCUSSION TITLE
RGB Mode handoff switch

### DISCUSSION THREAD

**Discussion by WhatMeansGithub** [basnijholt/adaptive-lighting#421] [Ideas]
I've recently switched out all of my warm/cool white bulbs to full RGB bulbs (livarno lux from Lidl). These bulbs can reproduce all colors well, better than my Ikea Tradfri bulbs did before, but the biggest issue is that when using RGB mode the colors are completely off (pink instead of orange/red). My current workaround is to have RGB mode unselected and just switch to a scene some time after sunset, but I think there's a better solution.

Now, I'm not a coder, but wouldn't it be possible to integrate a slider and toggle that lets you select at what color temperature the integration switches from normal color temp mode to RGB color temp mode? This way I could let my lights use the normal spectrum change until 2500K or something and once it hits that threshold the integration switches to RGB mode to get it down all the way to 1500K red (no issues with getting the red color). Would something like this be possible to add? Maybe even per bulb so you can select which bulb or which light groups use this mode?

**Comment by etchings13** [comment]
I've been trying to do something similar; for some reason my bulbs look better in RGB mode at night and color temp mode during the day. So I tried setting up an automation at sunset to switch to RGB mode using the "adaptive_lighting.apply" service with "prefer_rgb_color: true". It seems to work at first but for some reason always reverts back after a couple minutes and I don't know why. Might work for you though depending on what the issue is that's affecting my setup.

**Comment by th3w1zard1** [comment]
The issue with any implementation the devs make in `adaptive-lighting` is it would have to work with any bulb. All bulbs are calibrated differently, which is why `rgb_color` and `sleep_rgb_color` config options exist. Is that not enough configuration to solve your problem?

**Reply by WhatMeansGithub** [reply]
Unfortunately it isn’t. Like you said, each bulb is calibrated differently and the RGB color option therefore makes my lights look different. In my case instead of orange it gets pink. That’s why an offset or individual light calibration option would be very useful.

**Reply by th3w1zard1** [reply]
I don't quite understand how you change your light normally though?

You're saying the light gives you accurate colors outside the integration but don't say how you set it?

**Comment by roelanddewindt** [comment]
This is exactly the (underlying) issue I describe in https://github.com/orgs/home-assistant/discussions/3538. If we were able to correctly profile rgb led strips and bulbs, we could calibrate them to correctly reproduce any K value. This however seems not to be a something getting enough attention from the community...

EDIT: wrong link
